lib/bundler/installer/parallel_installer.rb

```diff
@@ -131,6 +135,12 @@ def handle_error
       raise Bundler::InstallError, errors.map(&:to_s).join("\n\n")
     end
 
+    def check_for_corrupt_lockfile
+      return unless @specs.any? {|s| s.corrupt_lockfile?(@specs) }
+      Bundler.ui.warn "Using 1 thread to install specs instead of #{@size} due to lockfile corruption" unless @size == 1
```

Comment:
Let's merge the warning in `corrupt_lockfile?` into this one, with wording something like:

> Your Gemfile.lock was created by an old Bundler that left some things out. Because of the missing `DEPENDENCIES`, we can only install gems one at a time, instead of installing #{@size} at a time.
> You can fix this by adding the missing gems to your `Gemfile`, running `bundle install`, and then removing the gems from your `Gemfile`. The missing gems are: #{missing.map(&:name).join(", ").